float32 t1_angle
float32 t1_radius
float32 t2_angle
float32 t2_radius
